tools/base/fpgaconf/bitstream-tools.h

```diff
@@ -165,6 +166,18 @@ fpga_result get_fpga_interface_id(fpga_token token, fpga_guid interface_id)
 		OPAE_ERR("Failed to get Token Properties Object \n");
 		goto out;
 	}
+
+	result = fpgaPropertiesGetObjectType(filter, &objtype);
+	if (result != FPGA_OK) {
+		OPAE_ERR("Failed to get Token Properties Object \n");
+		goto out_destroy;
+	}
+
+	if (objtype != FPGA_DEVICE) {
+		OPAE_ERR("Invalid FPGA object type \n");
```

Comment:
Thanks, Ananda.  Probably want to set result to an error code here, too. FPGA_EXCEPTION maybe?

Reply:
fixed